### PR TITLE
set default nixos version to 24.05

### PR DESCRIPTION
**Pull request by KiaraGrouwstra**

### CODE REVIEW

File: nix/hcloud/default.nix [hunk 1]
```diff
@@ -90,7 +90,7 @@ in
 
         provisioner.remote-exec = {
           inline = [
-            "curl https://raw.githubusercontent.com/elitak/nixos-infect/master/nixos-infect | PROVIDER=hetznercloud NIX_CHANNEL=nixos-23.11 NO_REBOOT=true bash 2>&1 | tee /tmp/infect.log"
```
Comment (aanderse) [inline nix/hcloud/default.nix:93]: i wonder if we should hold back with `hetzner` until [this](https://github.com/elitak/nixos-infect/issues/207) issue is triaged or resolved? :thinking:

**Review by aanderse — COMMENTED** [review]
thank you so much for your contribution!

only one comment, let me know what you think

**Comment by KiaraGrouwstra** [comment]
thanks for pointing that out! i've amended the PR accordingly.

File: examples/hcloud/flake.nix [hunk 2]
```diff
@@ -41,7 +41,7 @@
             location = "hel1";
           };
 
-          system.stateVersion = "23.11";
```
Comment (aanderse) [inline examples/hcloud/flake.nix:44]: sorry, i think we should revert this too

Comment (KiaraGrouwstra) [reply examples/hcloud/flake.nix]: done!

**Review by aanderse — APPROVED** [review]
:rocket:

thanks again for taking an interest!

**Pull request merged**